store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -5374,6 +5386,41 @@ private void authAccount(Account acct, String password,
 
     }
 
+    private void recoveryCodeBasedAuthAccount(Account acct, String recoveryCode, Map<String, Object> authCtxt)
+            throws ServiceException {
+        checkAccountStatus(acct, authCtxt);
+        if (!acct.getBooleanAttr(Provisioning.A_zimbraFeatureResetPasswordEnabled, false)) {
+            reportException(acct, "recovery code based auth can be used only when ResetPassword feature is enabled", authCtxt);
+        }
+
+        Map<String, String> codeMap = JWEUtil.getDecodedJWE(acct.getResetPasswordRecoveryCode());
+        if (codeMap == null) {
+            reportException(acct, "recovery code not found", authCtxt);
+        }
+
+        long expiryTime = Long.parseLong(codeMap.get(CodeConstants.EXPIRY_TIME.toString()));
```

Comment:
null check for codeMap is already there. If codeMap is null, exception is thrown.

Reply:
Yes but just because codeMap is not null does NOT mean it necessarily contains an entry for EXPIRY_TIME...  If that entry is missing then I'm pretty sure this code will NPE.

Reply:
added check.